api/dimensions.go

```diff
@@ -141,6 +147,13 @@ func (api *DatasetAPI) getDimensionOptions(w http.ResponseWriter, r *http.Reques
 	logData := log.Data{"dataset_id": datasetID, "edition": edition, "version": versionID, "dimension": dimension, "func": "getDimensionOptions"}
 	authorised := api.authenticate(r, logData)
 
+	versionId, err := checkVersion(ctx, versionID)
```

Comment:
use `versionID` instead of `versionId`, acronyms are always upper case unless they are unexported and variable starts with `id` - see https://betterprogramming.pub/naming-conventions-in-go-short-but-descriptive-1fa7c6d2f32a section on `Unique Names`

Please check all uses of this variable and change the name in other methods/functions if it doesn't abide by the go convention.

Reply:
Maybe the variable name would be better served as `versionNumber`?